src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,75 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  onStatusChanged: (e: string) => void,
+  onQueryChange: (e: string) => void,
+  status: string,
+  query: string,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
-      </span>
-    </p>
-  </form>
+export const TodoFilter: React.FC<Props> = React.memo(
+  ({
+    onStatusChanged,
+    status,
+    query,
+    onQueryChange,
+  }) => {
+    const handleStatusChange = (
+      event: React.ChangeEvent<HTMLSelectElement>,
+    ) => {
+      onStatusChanged(event.target.value);
+    };
+
+    const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      onQueryChange(event.target.value);
+    };
+
+    const clearButton = () => {
```

Comment:
Rename 'clearButton' function to 'handleClearButton' to maintain consistency with other event handler functions.